pymc3/step_methods/nuts.py

```diff
@@ -136,6 +137,14 @@ def astep(self, q0):
 
         return q
 
+    @staticmethod
+    def competence(var):
+        if var.dtype in continuous_types:
+            if sum(var.shape) > 5:
```

Comment:
One thing to note is that if a model has 10 continuous variables which are independently specified it will end up choosing metropolis not  NUTS. This is one reason why we might want to have the competence function take in *all* of the variables not consider them one at a time. I'm leaning in the direction that its better to do it this way.